File: public/docs/_examples/server-communication/ts/app/wiki/wikipedia.service.ts
// #docregion
import {Injectable} from 'angular2/core';
import {Jsonp, URLSearchParams} from 'angular2/http';

@Injectable()
export class WikipediaService {
  constructor(private jsonp: Jsonp) {}

  search (term: string) {

    let wikiUrl = 'http://en.wikipedia.org/w/api.php';

    // #docregion search-parameters
    var params = new URLSearchParams();
    params.set('search', term); // the user's search value
    params.set('action', 'opensearch');
    params.set('format', 'json');
    params.set('callback', 'JSONP_CALLBACK');
    // #enddocregion search-parameters

    // #docregion call-jsonp
    // TODO: Add error handling
    return this.jsonp
               .get(wikiUrl, { search: params })
               .map(request => <string[]> request.json()[1]);
    // #enddocregion call-jsonp
  }
}

File: public/docs/_examples/server-communication/dart/lib/wiki/wikipedia_service.dart
// #docregion
import 'dart:async';

import 'package:angular2/core.dart';
import 'package:jsonpadding/jsonpadding.dart';

@Injectable()
class WikipediaService {
  Future<List<String>> search(String term) async {
    // #docregion call-jsonp
    Uri uri = new Uri(
        scheme: 'http',
        host: 'en.wikipedia.org',
        path: 'w/api.php',
        queryParameters: {
          'search': term,
          'action': 'opensearch',
          'format': 'json'
        });
    // TODO: Error handling
    List result = await jsonp(uri);
    return result[1];
    // #enddocregion call-jsonp
  }
}
